policyenforcer.go

```diff
@@ -46,9 +46,33 @@ type ValidationReport struct {
 	ArtifactReports []*ValidationReport
 }
 
-// PolicyEnforcer is an interface with methods that make policy decisions.
+// Evaluator is an interface that defines methods to aggregate and evaluate
+// verification results generated for an artifact per validation request.
+type Evaluator interface {
+	// Pruned checks if the policy evaluation is completed for provided pair of
+	// subject and artifact. 
+	// It returns [ErrVerifierPruned] if the policy evaluation is completed for 
+	// the given verifier. 
+	// It returns [ErrArtifactPruned] if the policy evaluation is completed for 
+	// the given artifact regardless of verifiers.
+	Pruned(ctx context.Context, subjectDigest, artifactDigest, verifierName string) error
+
+	// AddResult adds the verification result of the subject against the
+	// artifact to the evaluator for further evaluation.
+	AddResult(ctx context.Context, subjectDigest, artifactDigest string, artifactResult *VerificationResult) error
+
+	// EvaluateArtifact evaluates aggregated verification results for the given
+	// artifact.
+	EvaluateArtifact(ctx context.Context, artifactDigest string) error
```

Comment:
It seems misleading. Calling `EvaluateArtifact` will not evaluate the artifact immediately but to signal the policy enforcer that there will be no more results for the artifact (or perhaps the subject).

Reply:
How about the method name `Seal`?

Reply:
Maybe `Commit` is better.